.github/workflows/main.yml

```diff
@@ -0,0 +1,44 @@
+name: ci
+on: [push, pull_request]
+
+jobs:
+  build_and_deploy:
+    env:
+        BASE_VERSION: 5.0.0.2012
+    runs-on: windows-latest
+    steps:
+    - uses: actions/checkout@v1
+    - name: Download Nuget.exe
+      uses: NuGet/setup-nuget@v1.0.5
+      with:
+        nuget-version: 5.4.0
+    - name: setup-msbuild
+      uses: microsoft/setup-msbuild@v1
+    - name: Restore & Build
+      run: |
+        cd Xamarin.Forms.Platform.GTK
+        Nuget.exe restore Xamarin.Forms.Platform.GTK.csproj
+        msbuild Xamarin.Forms.Platform.GTK.csproj /property:Configuration=Release
+    - name: Package and upload
+      if: ${{ github.event_name == 'push' }}
+      run: |
+        $buildConfiguration = "Release"
+        $date = get-date -format "yyyyMMdd-HHmm"
+        $hash = & git rev-parse --short HEAD
+        # Generates a version number with format "{XF.Core base version}--date{date}.git-{commit-id}"
+        $version = "$env:BASE_VERSION--date$date.git-$hash" 
+        
+        $fileName= "DotNetForms.Platform.GTK.$version.nupkg"
+        
+        # Replaces '$core_version$' and '$version$' in GTK nuspec file with XF.Core dependency version and version generated above for this commit
+        # Powershell scripts copied from https://github.com/xamarin/Xamarin.Forms/blob/5.0.0/build/steps/build-nuget.yml
+        Get-ChildItem './.nuspec/*.GTK.nuspec' -Recurse | Foreach-Object {
+             (Get-Content $_) | Foreach-Object  {
+                 $_ -replace '\$version\$', $version `
+                    -replace '\$core_version\$', $env:BASE_VERSION `
+                    -replace '\$Configuration\$', $buildConfiguration `
+             } | Set-Content $_
+        }
+
+        Nuget.exe pack './.nuspec/Xamarin.Forms.Platform.GTK.nuspec'
+        Nuget.exe push $fileName -ApiKey ${{secrets.NUGET_API_KEY}} -Source https://api.nuget.org/v3/index.json
```

Comment:
isn't `https://api.nuget.org/v3/index.json` the default source? are you sure that it needs to be specified?

Reply:
it made me to specify it I think you have to set it somewhere either nuget config file or args

Reply:
there's no Nuget.config file already in this repo?

Reply:
not sure haven't really looked I just tried without source and it didn't work so I added one

Reply:
it still sounds weird cause nugetPush.fsx doesnt use this flag I think

Reply:
ah, maybe it's because there is more than one source in the Nuget.config file